.ci/publishNpm.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+set -e
+
+if [ -z "$VERSION" ]; then
+    echo "The environment variable 'VERSION' must be set."
+    exit 1
+fi
+
+# set the version of all packages in the workspace to $VERSION
+npm version -ws $VERSION
+
+# replace all dependencies from the current workspace that are marked with "*" with the version that gets published
+find . -regex "./packages/[A-Za-z-]*/package\.json$" -exec sed -i -e "s/\"\*\"/\"$VERSION\"/g" {} \;
```

Comment:
This works by replacing all `*`s with $VERSION which is fine. If ever you should need a little more robust solution you can use the json parser `jq`, which can overwrite fields, too.

Reply:
It should replace `"*"` with `"<the-version>"` which should be quite robust. As we have more than one field to write (for example in runtime it's three) it would be many commands.